### PR TITLE
Adding some unit tests for the controller

### PR DESCRIPTION
**Pull request by sacharya**

### CODE REVIEW

File: pkg/controller/cincinnati/cincinnati_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	cincinnati := newDefaultCincinnati()
+	r := newTestReconciler(cincinnati)
+	request := newRequest(cincinnati)
+
+	_, err := r.Reconcile(request)
+	if err != nil {
+		t.Fatal(err)
+	}
+	instance := &cv1alpha1.Cincinnati{}
+	r.client.Get(nil, request.NamespacedName, instance)
```
Comment (pawanpinjarkar) [inline pkg/controller/cincinnati/cincinnati_controller_test.go:63]: Do not pass a nil Context, even if a function permits it; pass `context.TODO`

File: pkg/controller/cincinnati/cincinnati_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	cincinnati := newDefaultCincinnati()
+	r := newTestReconciler(cincinnati)
+
+	resources, err := newKubeResources(cincinnati, testOperandImage)
+	err = r.ensureConfig(nil, log, cincinnati, resources)
+	if err != nil {
+		t.Fatal(err)
+	}
+	found := &corev1.ConfigMap{}
+	err = r.client.Get(nil, types.NamespacedName{Name: nameConfig(cincinnati), Namespace: cincinnati.Namespace}, found)
```
Comment (pawanpinjarkar) [inline pkg/controller/cincinnati/cincinnati_controller_test.go:78]: Same as above. Please make the change for the remaining test cases.

File: pkg/controller/cincinnati/cincinnati_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	assert.True(t, conditionsv1.IsStatusConditionTrue(instance.Status.Conditions, cv1alpha1.ConditionReconcileCompleted))
+}
+
+// TestEnsureConfig ensures that Config is created successully and the OwnerReference is set.
+func TestEnsureConfig(t *testing.T) {
+	cincinnati := newDefaultCincinnati()
+	r := newTestReconciler(cincinnati)
+
+	resources, err := newKubeResources(cincinnati, testOperandImage)
+	err = r.ensureConfig(nil, log, cincinnati, resources)
```
Comment (pawanpinjarkar) [inline pkg/controller/cincinnati/cincinnati_controller_test.go:73]: Same as above.

**Comment by pawanpinjarkar** [comment]
Using `context.TODO` instead of `nil` in test cases safeguards us from build failures (at least) if we happen to implement the same code style/linting pattern as other repo like foundation in the future.

File: pkg/controller/cincinnati/cincinnati_controller_test.go (hunk 1 above)
Comment (sacharya) [reply pkg/controller/cincinnati/cincinnati_controller_test.go]: Sounds like a sensible default.

**Pull request merged**